client/task_runner.go

```diff
@@ -1061,8 +1072,12 @@ func (r *TaskRunner) run() {
 	}
 }
 
-// cleanup calls Driver.Cleanup when a task is stopping. Errors are logged.
+// cleanup removes Consul entries and calls Driver.Cleanup when a task is
```

Comment:
Great catch! Fixed and tested: https://github.com/hashicorp/nomad/pull/2467/commits/9625e19fe86620a552adb547adf4b7018c2c941d

Ensured test failed without the RemoveTask call when restarting.